src/core/net/dhcp6_client.cpp

```diff
@@ -513,9 +513,7 @@ void Dhcp6Client::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aM
     Dhcp6Header header;
     (void)aMessageInfo;
 
-    VerifyOrExit(aMessage.GetLength() - aMessage.GetOffset() >= static_cast<uint16_t>(sizeof(Dhcp6Header)), ;);
-
-    aMessage.Read(aMessage.GetOffset(), sizeof(header), &header);
+    VerifyOrExit(aMessage.Read(aMessage.GetOffset(), sizeof(header), &header) == sizeof(Dhcp6Header),);
```

Comment:
minor nit: sizeof(header) instead sizeof(Dhcp6Header) to keep same change along PR.